### PR TITLE
REAL CHANGES Issue 659/upgrade to midgard 2

### LINKED ISSUE #659: Upgrade to Midgard v.2.0
Doc: https://testnet.midgard.thorchain.info/v2/doc
JSON: https://testnet.midgard.thorchain.info/v2/swagger.json

Note: `v.2.0.0` includes breaking changes. One example is `thorchain/pool_addresses` -> `/thorchain/inbound_addresses` and (maybe) others.

### PR DESCRIPTION
**Pull request by thatStrangeGuyThorchain**
NOTE

For the sake of simplicity of review here are only meaning changes. all generated code for `v2` is at the #671 

closes #659

- [ ] refactor loadPoolsStateData$ to midgard 2.0
- [ ] some minor values which we can not get for now from `v2` were mocked as zero values (e.g. `poolTxAverage `)
- [ ] `StakersData` was removed from generated types. Created a domain-layer copy to have the same expected result for all stuff we already have
- [ ] refactor `getEnabledPool`/`getPendingPools `to avoid multiple requests for /api/pools
- [ ] getting `poolSlipAverage` and `swappingTxCount`  from `midgardApiV2.getSwapHistory` for poolDetails
- [ ] removed ALL usages of PoolDetail from generated to the custom Domain’s layer PoolDetail
- [ ] added liveData.altOnError
- [ ] made `getBlocksLeftForPendingPool` and `getBlocksLeftForPendingPoolAsString` to be dependent on pool’s asset

### CODE REVIEW

File: src/renderer/services/midgard/pools.ts [hunk 1]
```diff
@@ -59,71 +59,136 @@ const createPoolsService = (
...
+              }))
+            )
+          ),
+          liveData.sequenceArray,
+          liveData.map(
+            A.mapWithIndex((index, swapData) => ({
+              ...swapData,
+              // Here we can be sure in indexing 'cuz we have the same order
+              // as details 'cuz we build this array based on details themselves
+              ...details[index]
```
Comment (veado) [inline src/renderer/services/midgard/pools.ts:121]: _Just curious:_ What happened if the API will change the order of results?

File: src/renderer/services/midgard/pools.ts [hunk 2]
```diff
@@ -59,71 +59,136 @@ const createPoolsService = (
...
+  const apiGetPoolsData$: (assetOrAssets: string | string[]) => LiveData<Error, PoolDetails> = (assetOrAssets) => {
+    const assets = Array.isArray(assetOrAssets) ? assetOrAssets : [assetOrAssets]
+
+    return FP.pipe(
+      apiGetPoolsEnabled$,
+      liveData.map(A.filter((poolDetail) => assets.includes(poolDetail.asset))),
+      liveData.map(NEA.fromArray),
+      liveData.chain(
+        O.fold(
+          (): LiveData<Error, PoolDetails> => Rx.of(RD.failure(new Error('No pools available'))),
```
Comment (veado) [inline src/renderer/services/midgard/pools.ts:179]: Pls add `// TODO (@thatStrangeGuy | @veado) Add i18n`

**Review by veado — APPROVED** [review]
Great job! _Just a question + small thing ..._

File: src/renderer/services/midgard/pools.ts (hunk 1 above)
Comment (thatStrangeGuyThorchain) [reply src/renderer/services/midgard/pools.ts:121]: @veado nothing will happen. everything will work as expected.
what is happening here:
- we are going through all pool details' with simple `A.map`
- - for each pool we will request appropriate swaps history 
- - for all failed request we set mocked `RD.success` value. so we have `(RD.initial | RD.pending | RD.success)[]` (no `RD.failure`) to avoid failing to the error on the next step by sequencing with `liveData.sequenceArray`
- in other words we do Array.map `pool => shwapHistoryByPool` (order is the same).
- sequence `Observable<RD[]>` to the `Observable<[RD]>` with `liveData.sequenceArray` (it will just use `combineLatest`). this will not lead to any reorder
- as a result we have 1to1 match (by arrays' indexes) between `details` and `requested swaps history`
